front/app/components/admin/FormResults/FormResultsQuestion/SentimentQuestion/components/GroupedSentimentScore/utils.ts

```diff
@@ -0,0 +1,40 @@
+import { Answer, GroupedAnswer } from 'api/survey_results/types';
+
+import { calculateResponseCountForGroup, SentimentAnswers } from '../../utils';
+
+// transformGroupedAnswerUsableArray:
+// Description: Converts a grouped answer into a more usable array.
+export function transformGroupedAnswerUsableArray(
+  groupedAnswer: GroupedAnswer[],
+  groupKey: string
+): Answer[] {
+  // Map each entry to its count for the specified group
+  const answersForSpecificGroup = groupedAnswer.map((entry) => {
+    const groupAnswer = entry.groups.find((group) => group.group === groupKey);
+
+    return {
+      answer: entry.answer,
+      count: groupAnswer?.count || 0,
+    };
+  });
+
+  return answersForSpecificGroup;
+}
+
+// getAverageValue
+// Description: Helper function to calculate average sentiment
+export const getAverageValue = (
```

Comment:
Perfectly valid to do this on the front-end but wondering why we're not doing it on the BE like the rest of the averages? Or does it need to be more dynamic?

Reply:
I didn't want to add to your load already with this - but yes probably ideally this would be calculated on the BE :thinking: I'll add this to an extra ticket on our board, in case we can get to it :slightly_smiling_face: